clang/test/3C/root_cause.c

```diff
@@ -2,58 +2,85 @@
 
 // This test is unusual in that it checks for the errors in the code
 
-#include <stdlib.h>
+// unwritable-expected-warning@+2 {{0 unchecked pointers: Source code in non-writable file}}
+// expected-warning@+1 {{Unchecked pointer in parameter or return of external function malloc}}
+_Itype_for_any(T) void *malloc(unsigned long size) : itype(_Array_ptr<T>) byte_count(size);
 
+
+// unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
 void *x; // expected-warning {{1 unchecked pointer: Default void* type}}
 
 void test0() {
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   int *a;
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   char *b;
+  // unwritable-expected-warning@+2 {{0 unchecked pointers: Source code in non-writable file}}
+  // unwritable-expected-warning@+1 {{incompatible pointer types assigning to 'int *' from 'char *'}}
   a = b; // expected-warning {{2 unchecked pointers: Cast from char * to int *}}
 
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   int *c;
+  // unwritable-expected-warning@+1 {{expression result unused}}
   (char *)c; // expected-warning {{1 unchecked pointer: Cast from int * to char *}}
 
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   int *e;
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   char *f;
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
   f = (char *)e; // expected-warning {{2 unchecked pointers: Cast from int * to char *}}
 }
 
 void test1() {
   int a;
-  int *b;
+  // unwritable-expected-warning@+1 {{0 unchecked pointers: Source code in non-writable file}}
+  int *b; 
+  // unwritable-expected-warning@+1 {{ 0 unchecked pointers: Source code in non-writable file}}
   b = malloc(sizeof(int)); // expected-warning {{1 unchecked pointer: Bad pointer type solution}}
   b[0] = 1;
 
   union u {
+    // unwritable-expected-warning@+1 {{ 0 unchecked pointers: Source code in non-writable file}}
     int *a; // expected-warning {{1 unchecked pointer: Union or external struct field encountered}}
+    // unwritable-expected-warning@+1 {{ 0 unchecked pointers: Source code in non-writable file}}
     int *b; // expected-warning {{1 unchecked pointer: Union or external struct field encountered}}
   };
 
-  void (*c)(void);
+  void (*c)(void); // unwritable-expected-warning {{0 unchecked pointers: Source code in non-writable file}}
   c++; // expected-warning {{1 unchecked pointer: Pointer arithmetic performed on a function pointer}}
 
+    // unwritable-expected-warning@+2 {{ 0 unchecked pointers: Source code in non-writable file}}
```

Comment:
Please fix the indentation and remove the space before the number of unchecked pointers, here and in the rest of this file.